File: java/fibonacci.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class Fibonacci {
    public static long fibonacci(int n, long[] memo) {
        if (n <= 1) return n;
        if (memo[n] != 0) return memo[n];
        memo[n] = fibonacci(n - 1, memo) + fibonacci(n - 2, memo);
        return memo[n];
    }

    public static void main(String[] args) {
        long startTime = System.nanoTime();

        int limit = 10000;
        long[] memo = new long[limit];
        for (int i = 0; i < limit; i++) {
            fibonacci(i, memo);
        }

        long endTime = System.nanoTime() - startTime;
        double executionTimeMs = endTime / 1_000_000.0;

        System.out.println("Java execution time in ms:" + executionTimeMs);

        // Crear directorio si no existe
        File dir = new File("/app/salida_java");
        if (!dir.exists()) {
            dir.mkdirs();
        }

        // Guardar el tiempo en un archivo
        File outputFile = new File(dir, "salida_java.txt");
        try (FileWriter writer = new FileWriter(outputFile)) {
            writer.write("Java execution time in ms:" + executionTimeMs);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: javascript/fibonacci.js
const fs = require('fs');
const path = require('path');

// Crear el directorio si no existe
const outputDir = "/app/salida_js/";
if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
}

function fibonacci(n, memo = {}) {
    if (n in memo) return memo[n];
    if (n <= 1) return n;
    memo[n] = fibonacci(n - 1, memo) + fibonacci(n - 2, memo);
    return memo[n];
}

const startTime = performance.now();
for (let i = 0; i < 10000; i++) {
    fibonacci(i);
}
const endTime = performance.now() - startTime;

// Imprimir correctamente la ejecución
console.log(`JavaScript execution time in ms:${endTime}`);

// Guardar en el archivo dentro del contenedor
const outputFile = path.join(outputDir, "salida_js.txt");

fs.writeFileSync(outputFile, `JavaScript execution time in ms:${endTime}`);

